src/test/regress/sql/row_types.sql

```diff
@@ -29,13 +29,15 @@ INSERT INTO test VALUES (1,2), (1,3), (2,2), (2,3);
 -- multi-shard queries support row types
 SELECT (x,y) FROM test ORDER BY x, y;
 SELECT (x,y) FROM test GROUP BY x, y ORDER BY x, y;
+SELECT ARRAY[(x,(y,x)),(y,(x,y))] FROM test ORDER BY x, y;
```

Comment:
```SQL
 SELECT ARRAY[(x,(y,x)),(y,(x,y),x)] FROM test WHERE x != 2 ORDER BY x, y;
```

The following query works fine on Postgres, but  fails on this patch with Citus.  Can we make it work?

Also, what about ARRAY's inside ARRAY, such as ` ARRAY[ARRAY[(x,(y,x)), (x,(y,x))],ARRAY[(y,(x,y))]]`, do they work with this patch?

Reply:
Nested arrays work, will include test case. I was not aware that arrays could have varying shapes for RECORDs, will look into further

Reply:
Don't believe arrays with varying RECORD shape can be supported. `array_in` expects all its elements to take the same typmod as it has